runhouse/resources/functionals/mapper.py

```diff
@@ -159,7 +165,10 @@ def call_method_on_replica(job):
                 results.append(
                     list(
                         tqdm(
-                            executor.map(call_method_on_replica, jobs), total=len(jobs)
+                            executor.submit(
+                                call_method_on_replica, jobs, i != retries - 1
```

Comment:
what is going on here lol

Reply:
added a retry field to `call_module_on_replica` to handles exceptions differently depending on if it's retrying again or not.